pos_partner_sale_warning/static/src/js/ProductScreen.esm.js

```diff
@@ -0,0 +1,28 @@
+/** @odoo-module **/
+
+import PartnerListScreen from "point_of_sale.PartnerListScreen";
+import Registries from "point_of_sale.Registries";
+
+export const PartnerListScreenWarning = (PartnerListScreen) =>
+    class extends PartnerListScreen {
+        clickPartner(partner) {
+            if (
+                this.state.selectedPartner &&
+                this.state.selectedPartner.id === partner.id
+            ) {
+                return super.clickPartner(partner);
+            }
+            if (partner && partner.sale_warn && partner.sale_warn !== "no-message") {
+                this.showPopup("ErrorPopup", {
+                    title: this.env._t(`Warning for ${partner.name}`),
+                    body: this.env._t(partner.sale_warn_msg),
+                });
+                if (partner.sale_warn === "block") {
+                    return;
+                }
+            }
+            return super.clickPartner(partner);
+        }
+    };
+
+Registries.Component.extend(PartnerListScreen, PartnerListScreenWarning);
```

Comment:
File name should be adjusted

pos_partner_sale_warning/static/src/js/ProductScreen.esm.js ->
pos_partner_sale_warning/static/src/js/PartnerListScreen.esm.js

What do you think?

Reply:
Agree with you!